play/src/main/scala/caliban/PlayJson.scala

```diff
@@ -0,0 +1,123 @@
+package caliban
+
+import caliban.Value._
+import caliban.parsing.adt.LocationInfo
+import play.api.http.Writeable
+import play.api.libs.json.Reads._
+import play.api.libs.json.Writes._
+import play.api.libs.json._
+
+import scala.util.Try
+
+object PlayJson {
+
+  //// JSON Reads
+  private def jsonToInputValue(json: JsValue): InputValue = json match {
+    case JsNull             => NullValue
+    case boolean: JsBoolean => BooleanValue(boolean.value)
+    case JsNumber(value) =>
+      Try(value.toIntExact).fold(_ => FloatValue(value), IntValue(_))
```

Comment:
I'm actually not sure if this conversion is always true, but `Try` seems kind of expensive if it is a `Float` since it will have to capture a thrown `Exception` seems like it could just split this into two case blocks:

```scala
case JsNumber(value) if value.isWhole => IntValue(value.intValue)
case JsNumber(value) => FloatValue(value)
```

Reply:
Great suggestion! But I think checking only with `isWhole` is not safe, I could use `isValidXXX` methods alongside.